### PR TITLE
allowed to config enforce_eager.

### PR DESCRIPTION
**Pull request by adoda**

### CODE REVIEW

**Review by haolin-nju — DISMISSED** [review]
LGTM

**Review by charles9304 — APPROVED** [review]

File: chatlearn/models/vllm_module.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def _init_args(self):
             max_num_batched_tokens=self.model_args.get("max_num_batched_tokens"),
             max_num_seqs=self.model_args.get("micro_batch_size"),
             max_model_len=self.model_args.get("seq_length"),
-            enforce_eager=True,
+            enforce_eager=self.model_args.get("enforce_eager", False),
```
Comment (haolin-nju) [inline chatlearn/models/vllm_module.py:134]: the default value is inconsistent between *.py and *.yaml?

**Review by haolin-nju — COMMENTED** [review]
See comments

File: examples/megatron/configs/llama2/vllm_policy_inference.yaml [hunk 2]
```diff
@@ -29,6 +29,9 @@ max_num_batched_tokens: ${max_num_batched_tokens:32768}
 max_paddings: ${max_paddings:512}
 num_scheduler_steps: ${num_scheduler_steps:1}
 
+# engine config
+enforce_eager: ${enforce_eager:True}
```
Comment (charles9304) [inline examples/megatron/configs/llama2/vllm_policy_inference.yaml:33]: title should change as well, which shows "default=False"

File: chatlearn/models/vllm_module.py (hunk 1 above)
Comment (adoda) [reply chatlearn/models/vllm_module.py:134]: enforce_eager=False in .py is used to test for qwen, enforce_eager=True for llama.

File: examples/megatron/configs/llama2/vllm_policy_inference.yaml (hunk 2 above)
Comment (adoda) [reply examples/megatron/configs/llama2/vllm_policy_inference.yaml:33]: done

**Review by charles9304 — DISMISSED** [review]

**Review by haolin-nju — COMMENTED** [review]
LGTM

**Review by haolin-nju — DISMISSED** [review]

File: chatlearn/models/vllm_module.py (hunk 1 above)
Comment (haolin-nju) [reply chatlearn/models/vllm_module.py:134]: Configs for different models should be set in *.yaml. `enforce_eager` in *.py should be set as True in my opinion.

**Review by charles9304 — APPROVED** [review]

**Review by haolin-nju — APPROVED** [review]

**Pull request merged**